gen/dcos-config.yaml

```diff
@@ -407,9 +407,20 @@ package:
   - path: /etc/dcos-log.env
     content: |
       DCOS_LOG_CONFIG_PATH=/opt/mesosphere/etc/dcos-log-config.json
-  - path: /etc/dcos-log-config.json
+  - path: /etc_master/dcos-log-config.json
     content: |
       {
+        "role": "master"
+      }
+  - path: /etc_slave/dcos-log-config.json
+    content: |
+      {
+        "role": "agent"
+      }
+  - path: /etc_slave_public/dcos-log-config.json
+    content: |
+      {
+        "role": "agent_public"
```

Comment:
here is when i pass role to a library function https://github.com/dcos/dcos-log/blob/master/dcos-log/api/server.go#L32

later in middleware i call this function https://github.com/dcos/dcos-log/blob/master/dcos-log/api/middleware.go#L137

which eventually will do this https://github.com/dcos/dcos-go/blob/master/dcos/nodeutil/util.go#L282